.github/workflows/ci.yml

```
@@ -53,10 +53,16 @@ jobs:
         shell: bash
         run: echo "INSTA_SNAPSHOT_REFERENCES_FILE=$(mktemp)" >> "${GITHUB_ENV}"
 
-      - name: Run the test suite
+      - name: Run the test suite (Without AWS KMS)
+        if: ${{ runner.os != 'Linux' }}
         shell: bash
         run: cargo test --timings --workspace --locked
 
+      - name: Run the test suite (With AWS KMS)
+        if: ${{ runner.os == 'Linux' }}
+        shell: bash
+        run: cargo test --timings --workspace --locked --features aws-kms
```

I think we might need `--test-threads=1` here? 

```suggestion --
        run: cargo test --timings --workspace --locked --features aws-kms -- --test-threads=1
```